src/import_edf.jl

```diff
@@ -107,14 +107,15 @@ function edf_signal_encoding(edf_signal_header::EDF.SignalHeader,
     sample_resolution_in_unit = (pmax - pmin) / (dmax - dmin)
     sample_offset_in_unit = pmin - (sample_resolution_in_unit * dmin)
     sample_rate = edf_signal_header.samples_per_record / edf_seconds_per_record
+    sample_type = (dmax > typemax(Int16) || dmin < typemin(Int16)) ? Int32 : Int16
     return (sample_resolution_in_unit=Float64(sample_resolution_in_unit),
             sample_offset_in_unit=Float64(sample_offset_in_unit),
-            sample_rate=Float64(sample_rate))
+            sample_rate=Float64(sample_rate),
+            sample_type=sample_type)
 end
 
-# TODO: implement a more disciplined widening protocol for `sample_type`
 function promote_encodings(encodings; pick_offset=(_ -> 0.0), pick_resolution=minimum)
-    sample_type = Int16
+    sample_type = reduce(promote_type, (e.sample_type for e in encodings))
```

Comment:
I'll call out that `promote_type` takes varargs:
```suggestion
    sample_type = promote_type((e.sample_type for e in encodings)...)
```

Reply:
Yeah, but I prefer not to splat unless i'm in a situation where the splatted element count is known locally (here it's not - it's caller driven)